x-pack/plugins/enterprise_search/public/applications/shared/kibana_breadcrumbs/set_breadcrumbs.test.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { SetAppSearchBreadcrumbs } from '../kibana_breadcrumbs';
+import { mount } from 'enzyme';
+
+jest.mock('./generate_breadcrumbs', () => ({
+  appSearchBreadcrumbs: jest.fn(),
+}));
+import { appSearchBreadcrumbs } from './generate_breadcrumbs';
+
+jest.mock('react-router-dom', () => ({
+  useHistory: () => ({
+    createHref: jest.fn(),
+    push: jest.fn(),
+    location: {
+      pathname: '/current-path',
+    },
+  }),
+}));
+
+import { KibanaContext } from '../..';
+
+describe('SetAppSearchBreadcrumbs', () => {
+  const setBreadcrumbs = jest.fn();
+  const builtBreadcrumbs = [];
+  const appSearchBreadCrumbsInnerCall = jest.fn().mockReturnValue(builtBreadcrumbs);
+  const appSearchBreadCrumbsOuterCall = jest.fn().mockReturnValue(appSearchBreadCrumbsInnerCall);
+  appSearchBreadcrumbs.mockImplementation(appSearchBreadCrumbsOuterCall);
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const render = props => {
+    return mount(
+      <KibanaContext.Provider
+        value={{
+          http: {},
+          enterpriseSearchUrl: 'http://localhost:3002',
+          setBreadcrumbs,
+        }}
+      >
+        <SetAppSearchBreadcrumbs {...props} />
+      </KibanaContext.Provider>
+    );
+  };
+
+  describe('when isRoot is false', () => {
+    const subject = () => render({ text: 'Page 1', isRoot: false });
+
+    it('calls appSearchBreadcrumbs to build breadcrumbs, then registers them with Kibana', () => {
+      subject();
+
+      // calls appSearchBreadcrumbs to build breadcrumbs with the target page and current location
+      expect(appSearchBreadCrumbsInnerCall).toHaveBeenCalledWith([
+        { text: 'Page 1', path: '/current-path' },
+      ]);
+
+      // then registers them with Kibana
+      expect(setBreadcrumbs).toHaveBeenCalledWith(builtBreadcrumbs);
+    });
+  });
+
+  describe('when isRoot is true', () => {
```

Comment:
I like the unit tests in this file. From reading the tests alone, I get a sense of what `<SetSearchBreadCrumbs/>` is intended to do: take in some breadcrumbs, and set them in Kibana. It makes it easy to test the behavior of the component without having to load all of Kibana. It verifies that the external touchpoints (e.g. Kibana's `setBreadCrumbs` API) are called.

I'm not exactly sure _why_ when `isRoot` is true we build breadcrumbs with an empty breadcrumb, but I imagine the docs for the component itself would explain why this is a useful and necessary thing to do.

Reply:
And... it's clear to me now after having read the implementation of `SetAppSearchBreadcrumbs` why `isRoot` works the way it does. Cool.

Reply:
Sorry, I think this was bad prop naming on my part - I kind of hand-waved a flag that I thought sounded good for "When the breadcrumb is just the 'App Search' base") but I think that could be cleaned up significantly.

My original goal was to make it easy to do nested breadcrumbs, like so

```jsx
appSearchBreadcrumbs(history)([
  { text: 'foo', path: '/foo' },
  { text: 'bar', path: '/foo/bar' },
]);
```

but I'm not totally sure I succeeded since it made setting a base App Search path (aka `isRoot`) with no child breadcrumbs a little opaque. 😬 

I'm definitely open to rewriting the source code here!